### PR TITLE
Fix/inline edit task name

### PR DESCRIPTION
**Pull request by WINt1703**
fix: #450

### CODE REVIEW

File: app/frontend/components/taskList/taskList.tsx [hunk 1]
```diff
@@ -7,7 +7,7 @@ import { ErrorMessage } from '@hookform/error-message'
 import { DeleteTaskModal } from '../deleteTaskModal'
 import { useState } from 'react'
 import { yupResolver } from '@hookform/resolvers/yup'
-import { TaskDetailsModal, taskInputSchema } from '../../../frontend/components/taskDetailsModal'
+import { taskInputSchema } from '../../../frontend/components/taskDetailsModal'
```
Comment (PascalHelbig) [inline app/frontend/components/taskList/taskList.tsx:10]: Can we delete the `TaskDetailsModal` component, or is it still used elsewhere?

File: app/frontend/components/taskList/taskRow.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        id: task.id,
+        data: {
+          projectId: task.project.id,
+          title: taskData.title,
+        },
+      })
+      if (result.error) {
+        throw new Error(`GraphQL Error ${result.error}`)
+      }
+      reset()
```
Comment (PascalHelbig) [inline app/frontend/components/taskList/taskRow.tsx:41]: Can you explain why we need the `reset()` here? I can not figure it out.

File: app/frontend/components/taskList/taskRow.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error(error)
+    }
+  }
+
+  return (
+    <div className="flex w-[500px]">
```
Comment (PascalHelbig) [inline app/frontend/components/taskList/taskRow.tsx:49]: I would prefer min/max width.
While we don't have a responsive design, I think we should not use fixed widths where we can.

File: app/frontend/components/taskList/taskRow.tsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+      <InputField
+        variant="primary"
+        className="font-small dark:placeholder-grey ml-2 rounded read-only:bg-gray-100 read-only:opacity-50 dark:border-white dark:bg-slate-800 dark:text-white"
```
Comment (PascalHelbig) [inline app/frontend/components/taskList/taskRow.tsx:52]: Do we need those tailwind classes?
I would let the `InputField` component do the styling.

File: app/frontend/components/taskList/taskRow.tsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        >
+          <path
+            d="M100 50.5908C100 78.2051 77.6142 100.591 50 100.591C22.3858 100.591 0 78.2051 0 50.5908C0 22.9766 22.3858 0.59082 50 0.59082C77.6142 0.59082 100 22.9766 100 50.5908ZM9.08144 50.5908C9.08144 73.1895 27.4013 91.5094 50 91.5094C72.5987 91.5094 90.9186 73.1895 90.9186 50.5908C90.9186 27.9921 72.5987 9.67226 50 9.67226C27.4013 9.67226 9.08144 27.9921 9.08144 50.5908Z"
+            fill="currentColor"
+          />
+          <path
+            d="M93.9676 39.0409C96.393 38.4038 97.8624 35.9116 97.0079 33.5539C95.2932 28.8227 92.871 24.3692 89.8167 20.348C85.8452 15.1192 80.8826 10.7238 75.2124 7.41289C69.5422 4.10194 63.2754 1.94025 56.7698 1.05124C51.7666 0.367541 46.6976 0.446843 41.7345 1.27873C39.2613 1.69328 37.813 4.19778 38.4501 6.62326C39.0873 9.04874 41.5694 10.4717 44.0505 10.1071C47.8511 9.54855 51.7191 9.52689 55.5402 10.0491C60.8642 10.7766 65.9928 12.5457 70.6331 15.2552C75.2735 17.9648 79.3347 21.5619 82.5849 25.841C84.9175 28.9121 86.7997 32.2913 88.1811 35.8758C89.083 38.2158 91.5421 39.6781 93.9676 39.0409Z"
+            fill="currentFill"
+          />
+        </svg>
```
Comment (PascalHelbig) [inline app/frontend/components/taskList/taskRow.tsx:71]: Can we use here an icon from `react-icons` package?

File: app/frontend/components/taskList/taskList.tsx (hunk 1 above)
Comment (WINt1703) [reply app/frontend/components/taskList/taskList.tsx:10]: yes, we use ```yup resolver``` for validation field

File: app/frontend/components/taskList/taskRow.tsx (hunk 4 above)
Comment (WINt1703) [reply app/frontend/components/taskList/taskRow.tsx:52]: because ```InputField``` doesn't support dark theme

File: app/frontend/components/taskList/taskList.tsx (hunk 1 above)
Comment (PascalHelbig) [reply app/frontend/components/taskList/taskList.tsx:10]: Okay, then I would say let's put the validation in this file and delete the other one.
We don't need any unused code.

File: app/frontend/components/taskList/taskRow.tsx (hunk 4 above)
Comment (PascalHelbig) [reply app/frontend/components/taskList/taskRow.tsx:51]: But then the `InputField` component would be the right place to fix that.
We have the component to keep this somewhat consistent. 😊

Comment (WINt1703) [reply app/frontend/components/taskList/taskRow.tsx:49]: i checked all places where we use ```InputField``` and everywhere set colors for dark theme

Comment (moncapitaine) [reply app/frontend/components/taskList/taskRow.tsx:49]: I think Pascal wanted the className values inside the InputField component and not setting it in usage of InputField

File: app/frontend/components/taskList/taskRow.tsx [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    <div className="flex min-w-[500px]">
+      <InputField
+        variant="primary"
+        className="ml-2"
+        {...register('title', { required: true })}
+        onBlur={handleSubmit(handleSubmitTask)}
+      />
+
+      {fetching && <CgSpinner className="ml-2 inline h-8 w-8 animate-spin dark:text-blue-600" />}
+      <ErrorMessage errors={errors} name="title" as={<span aria-label="error field" className="ml-2 text-red-700" />} />
```
Comment (PascalHelbig) [inline app/frontend/components/taskList/taskRow.tsx:55]: @WINt1703, do you need the `aria-label` to make this span accessible in your test?
This is not what aria-label is for.

> The `aria-label` attribute defines a string value that labels an interactive element.
> https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label

But you can use the `data-testid` attribute in testing library instead:
https://testing-library.com/docs/queries/bytestid/

File: app/frontend/components/taskList/taskRow.tsx [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+          projectId: task.project.id,
+          title: taskData.title,
+        },
+      })
+      if (result.error) {
+        throw new Error(`GraphQL Error ${result.error}`)
+      }
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error(error)
```
Comment (PascalHelbig) [inline app/frontend/components/taskList/taskRow.tsx:41]: What will happen when the mutation fails?

Will the input have the original text again? Maybe it is worth to write a test for this edge case.

File: app/frontend/components/taskList/taskRow.tsx (hunk 6 above)
Comment (moncapitaine) [reply app/frontend/components/taskList/taskRow.tsx:55]: use role alert for this: (https://www.w3.org/WAI/GL/wiki/Using_ARIA_role_of_alert_for_Error_Feedback_in_Forms)

File: app/frontend/components/taskList/taskRow.tsx [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+      id: task.id,
+      data: {
+        projectId: task.project.id,
+        title: taskData.title,
+      },
+    })
+  }
+
+  return (
+    <div className="flex min-w-[500px]">
```
[the remark this answers is not in the document]
Comment (WINt1703) [reply app/frontend/components/taskList/taskRow.tsx:38]: Min width need to error message

[the remark this answers is not in the document]
Comment (WINt1703) [reply app/frontend/components/taskList/taskRow.tsx:38]: You can clear cell and check message

File: app/frontend/components/taskList/taskList.tsx [hunk 9]
```diff
@@ -79,27 +82,13 @@ export const TaskList = (props: TaskListProps): JSX.Element => {
                     <BiTrash />
                   </Button>
                 )}
-                <span className="ml-2">{task.title}</span>
+                <TaskRow task={task} />
```
[the remark this answers is not in the document]
Comment (PascalHelbig) [reply app/frontend/components/taskList/taskList.tsx]: Hey @WINt1703, why is this comment resolved?
You didn't even respond to my suggestion. Did you miss it or not understand it?

File: app/frontend/components/taskList/taskRow.tsx (hunk 7 above)
Comment (WINt1703) [reply app/frontend/components/taskList/taskRow.tsx]: i removed try catch

File: app/frontend/components/taskList/taskCell.tsx [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+        title: taskData.title,
+      },
+    })
+
+    if (result.error) setError('title', { message: 'Network error' })
+  }
+
+  return (
+    <>
+      {canModify && (
```
Comment (moncapitaine) [inline app/frontend/components/taskList/taskCell.tsx:47]: rename to canDelete
